database.py

```diff
@@ -0,0 +1,53 @@
+"""
+Contains definitions of database models.
+"""
+
+
+from flask_sqlalchemy import SQLAlchemy
+
+db = SQLAlchemy()
+
+
+class BTUser(db.Model):
+    """Database model for BotTrition users."""
+
+    __tablename__ = "btuser"
+
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(80), unique=True, nullable=False)
+    password_hash = db.Column(db.String(80), nullable=False)
+
+    restrictions = db.relationship("DietaryRestriction", backref="btuser")
+    profile = db.relationship("Profile", backref="btuser", uselist=False)
+
+
+# So basically the Allergies and DietaryRestriction are multivalued so our user will have
+# one to many relationship with these two attributes.
+
+
+class Profile(db.Model):
+    """Database model for BotTrition user profile information."""
+
+    __tablename__ = "profile"
+
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey("btuser.id"))
+
+    gender = db.Column(db.String(80))
+    height = db.Column(db.Integer)
```

Comment:
I see why this is an integer to have an easier time calculating for the BMI. However, what about the possibility for this to be a string rather than an integer, because a user maybe 6 feet tall but might not know the conversion to add 72inches in the table to put as their height. Some issues of switching to a string will be having to get the height from the string.

Reply:
I think we should perform that conversion on the user's input. The frontend could offer two separate fields for the feet and inches components of height. For example, two textboxes separated by `'` and `"` with placeholder text could signal to the user that feet and inches are expected. Then, we could convert the height to a single integer before storing it.